latex/main.tex

```diff
@@ -55,12 +47,85 @@
+      \hline
+      \makecell{No frozen layers} & \makecell{0.9856} & \makecell{0.0} & \makecell{2}\\ 
+      \makecell{Critical layer (1)} & \makecell{0.9751} & \makecell{0.0105}& \makecell{17}\\
+      \makecell{Single ambient layer\\ 
+      (2,3,4 average) }
+        & \makecell{0.9829} & \makecell{0.002}& \makecell{2}\\
+      \makecell{All ambient layers\\ (2,3,4)} &  \makecell{0.9774} & \makecell{0.0082} & \makecell{53}
+      
+    \end{tabular}
+\caption{Test accuracy on 3 layered fully connected network on MNIST, by frozen layers during training.} 
+  \end{center}
+\end{table}
+
+\begin{figure}
+  \includegraphics[width=\linewidth]{images/fc3_mnist_training_plots.png}
+  \caption{Test accuracy by epoch, for different sets of layers frozen during setting. TODO elaborate: error bars.}
+  \label{fig:fc3_training}
+\end{figure}
+\section{Robustness with frozen layers}
+We examine the robustness effect when training with frozen layers. We trained a fully connected network, while freezing a layer (or a set of layers) throughout training. Then we apply re-initialization and re-randomization to measure robustness. We compare all the results to the robustness of a baseline model trained without freezing any layers.\\
+When training a baseline fully connected network with 3 layers on MNIST (figure \ref{fig:baseline_fc3_heatmap}), it is notable the first layer is critical while the other layers are ambient, and all layers are sensitive to re-randomization as in \cite{allLayers}.\\
+Another interesting phenomenon is that freezing layers during training changes the effect in different ways corresponding to the frozen layer's robustness (figure \ref{fig:fc3_drop_by_layer_type}):\\ 
+\begin{figure}
+  \includegraphics[width=\linewidth]{images/fc3_mnist_drop_in_acc_by_layer.png}
+  \caption{Robustness by frozen layers and reinitialized layer type. Drop in test accuracy for each model trained with a different frozen layer (columns), when the corresponding type of layer (rows) is re-initialized to epoch 0. For ambient layers, the average is presented, with STD $<$ 0.001 for all set of layers. (TODO: validate!)}
+  \label{fig:fc3_drop_by_layer_type}
+\end{figure}
+\emph{Freezing a critical layer:} When freezing the first layer, the proceeding layer "substitutes" for it and becomes critical in larger factor. Re-initializing the first layer on the baseline model drops test accuracy from 98.5\% to 54\%. On the first layer frozen model, re-initializing the second layer demonstrates a larger decrease in test accuracy: from 97.5\% to 45\%. Not only the effect is stronger, but it lasts for more epochs too: on the baseline model, the effect almost vanishes when resetting the first layer to the second epoch weights (98.5\% drops to 97.7\%). When the first layer is frozen, resetting the second layer weights to the second epoch checkpoint shows a larger drop in performance (97.5\% drops to 95.5\%).\\
```

Comment:
"...and becomes critical in larger factor." I think this should be rephrased to make it clear what the "large factor" is describing; i.e. "...and becomes more critical than the first layer was before freezing"